provider/azure/environ_test.go

```diff
@@ -1202,6 +1203,29 @@ func (s *environSuite) TestStopInstancesNotFound(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *environSuite) TestStopInstancesResourceGroupNotFound(c *gc.C) {
+	// skip storage, so we get to deleting security rules
+	s.PatchValue(&s.storageAccountKeys.Keys, nil)
+	env := s.openEnviron(c)
+
+	nsgSender := s.makeSender(".*/networkSecurityGroups/juju-internal-nsg", makeSecurityGroup())
+	nsgSender.SetError(autorest.NewErrorWithError(errors.New("autorest/azure: Service returned an error."), "network.SecurityGroupsClient", "Get", &http.Response{StatusCode: http.StatusNotFound}, "Failure responding to request"))
+
+	s.sender = azuretesting.Senders{
+		s.makeSender("/deployments/machine-0", s.deployment), // Cancel
+		s.storageAccountSender(),
+		s.storageAccountKeysSender(),
+		s.networkInterfacesSender(),                       // GET: no NICs
+		s.publicIPAddressesSender(),                       // GET: no public IPs
+		s.makeSender(".*/virtualMachines/machine-0", nil), // DELETE
+		s.makeSender(".*/disks/machine-0", nil),           // DELETE
+		nsgSender, // GET with failure
+		s.makeSender(".*/deployments/machine-0", nil), // DELETE
+	}
+	err := env.StopInstances("machine-0")
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
we also need to check call names like is done elsewhere

Reply:
this area is not setup for checking the call tree.  however i did verify with log messages that we hit the correct section of code, as well as a different http.Status which would cause a failure.